internal/experiments/chaos_controller/chaosmesh.go

```diff
@@ -0,0 +1,232 @@
+	_, err = cc.dynamicClient.Resource(gvr).Namespace(cc.namespace).Create(context.TODO(), parsedYaml, v1.CreateOptions{})
+	if err != nil {
+		log.Fatalf("Failed to apply ChaosMesh resource: %v", err)
+	}
+
+	cc.ExperimentSleep(recoveryTime)
+
+	if restartNodes {
+		RestartAffectedPods(cc)
+	}
+	return err
+}
+
+// Deleting the experiment from chaosmesh list
+// to avoid collisions, when the same experiment is started
+func (cc *ChaosController) ClearExperimentData(chaosType string) {
+	gvr := schema.GroupVersionResource{
+		Group:    "chaos-mesh.org",           // ChaosMesh group
+		Version:  "v1alpha1",                 // Version of the ChaosMesh resource
+		Resource: strings.ToLower(chaosType), // Resource is "podchaos"
+	}
+
+	err := cc.dynamicClient.Resource(gvr).Namespace(cc.namespace).DeleteCollection(
+		context.TODO(),
+		v1.DeleteOptions{},
+		v1.ListOptions{},
+	)
+	if err != nil {
+		log.Fatalf("Failed to delete PodChaos resources: %v", err)
+	}
+}
+
+// Check the health of each node in cluster
+// Returning string so that JS can handle it correctly
+func (cc *ChaosController) CheckClusterHealth() string {
+	allPods, err := cc.pod.List(context.Background(), cc.namespace)
+	if err != nil {
+		return err.Error()
+	}
+
+	err = cc.CheckPodsHealth(allPods)
+	if err != nil {
+		return err.Error()
+	}
+	return ""
+
+}
+
+// Check the status of pods, affected by the chaos experiment
+func (cc *ChaosController) CheckAffectedPodsLiveness() string {
+	affectedNodes, ok, err := unstructured.NestedStringSlice(cc.curExp.Object, "spec", "selector", "pods", extractNamespace(cc.curExp))
+	if !ok {
+		return err.Error()
+	}
+
+	err = cc.CheckPodsHealth(affectedNodes)
+	if err != nil {
+		return err.Error()
+	}
+	return ""
+}
+
+func (cc *ChaosController) CheckPodsHealth(pods []string) error {
+	cur_namespace := extractNamespace(cc.curExp)
+	for _, podName := range pods {
+		err := CheckSingleNodeHealth(cc.pod.Client, cur_namespace, podName)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// Check node health, params are suited for go tests
+// Use k8s interface to pass both kubernetes.client and fake.client
+func CheckSingleNodeHealth(clientset kubernetes.Interface, namespace, podName string) error {
+	if clientset == nil {
+		return fmt.Errorf("clientset is nil")
+	}
+
+	pod, err := clientset.CoreV1().Pods(namespace).Get(context.TODO(), podName, v1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	for _, condition := range pod.Status.Conditions {
+		if condition.Type == coreV1.PodConditionType(coreV1.NodeReady) {
+			if condition.Status != coreV1.ConditionTrue {
+				return fmt.Errorf("node %s is not ready. Status: %s, Reason: %s, Message: %s",
+					podName, condition.Status, condition.Reason, condition.Message)
+			}
+		}
+	}
+
+	return nil
+}
+
+// Get all podes, affected by chaos experiment and reload them
+// This is done in order to avoid CrashLoopBackOff pod status
+func RestartAffectedPods(cc *ChaosController) error {
```

Comment:
Move the receiver before function name
```suggestion
func (cc *ChaosController) RestartAffectedPods() error {
```